code/modules/shuttle/marine_dropship.dm

```diff
@@ -976,48 +1000,128 @@
+		return TRUE
+
+	if(!M.can_move_topic(usr))
+		return TRUE
+
+	if(!params["destination"])
+		return TRUE
+
+	if(!(params["destination"] in valid_destinations()))
+		log_admin("[key_name(usr)] may be attempting a href dock exploit on [src] with target location \"[params["destination"]]\"")
+		message_admins("[ADMIN_TPMONTY(usr)] may be attempting a href dock exploit on [src] with target location \"[params["destination"]]\"")
+		return TRUE
+
+	var/previous_status = M.mode
+	log_game("[key_name(usr)] has sent the shuttle [M] to [params["destination"]]")
+
+	switch(SSshuttle.moveShuttle(shuttleId, params["destination"], 1))
+		if(0)
+			if(previous_status != SHUTTLE_IDLE)
+				visible_message("<span class='notice'>Destination updated, recalculating route.</span>")
+			else
+				visible_message("<span class='notice'>Shuttle departing. Please stand away from the doors.</span>")
+			return TRUE
+		if(1)
+			to_chat(usr, "<span class='warning'>Invalid shuttle requested.</span>")
+			return TRUE
+		else
+			to_chat(usr, "<span class='notice'>Unable to comply.</span>")
+			return TRUE
+
+/obj/machinery/computer/shuttle/shuttle_control/ui_data(mob/user)
+	var/list/data = list()
 	var/list/options = valid_destinations()
-	var/obj/docking_port/mobile/marine_dropship/M = SSshuttle.getShuttle(shuttleId)
-	var/dat = "Status: [M ? M.getStatusText() : "*Missing*"]<br><br>"
-	if (M?.hijack_state == HIJACK_STATE_NORMAL)
-		var/destination_found
+	var/obj/docking_port/mobile/M = SSshuttle.getShuttle(shuttleId)
+	if(!M)
+		return data //empty but oh well
+
+	data["linked_shuttle_name"] = M.name
+	data["shuttle_status"] = M.getStatusText()
+	for(var/option AS in options)
 		for(var/obj/docking_port/stationary/S in SSshuttle.stationary)
-			if(!options.Find(S.id))
-				continue
-			if(!M.check_dock(S, silent=TRUE))
+			if(option != S.id)
 				continue
-			destination_found = TRUE
-			dat += "<A href='?src=[REF(src)];move=[S.id]'>Send to [S.name]</A><br>"
-		if(!destination_found)
-			dat += "<B>Shuttle Locked</B><br>"
+			var/list/dataset = list()
+			dataset["id"] = S.id
+			dataset["name"] = S.name
+			dataset["locked"] = !M.check_dock(S, silent=TRUE)
+			data["destinations"] += list(dataset)
+	return data
+
+/// Relinks the shuttleId in the console to a valid shuttle currently existing. Will only relink to a shuttle with a matching control_flags flag. Returns true if successfully relinked
+/obj/machinery/computer/shuttle/shuttle_control/proc/RelinkShuttleId(forcedId)
+	var/newId = null
+	/// The preferred shuttleId to link to if it exists.
+	var/preferredId = initial(shuttleId)
+	var/obj/docking_port/mobile/M
+	var/shuttleName = "Unknown"
+	if(forcedId)
+		M = SSshuttle.getShuttle(forcedId)
+		if(!M)
+			return FALSE
+		newId = M.id
+		shuttleName = M.name
+	else
+		M = null
+		for(M in SSshuttle.mobile)
+			if(M.control_flags & SHUTTLE_MARINE_PRIMARY_DROPSHIP)
+				newId = M.id
+				shuttleName = M.name
```

Comment:
Early continue . Else if you forget to set M control flag to shuttle_marine_primary_dropship but is preferredId it breaks